astropy/units/equivalencies.py

```diff
@@ -24,22 +26,71 @@
            'pixel_scale', 'plate_scale', 'with_H0']
 
 
+class Equivalency(UserList):
+    """
+    A container for a units equivalency.
+
+    Attributes
+    ----------
+    name: `str`
+        The name of the equivalency.
+    args: `tuple`
```

Comment:
For (future) introspection, it might be more useful only to have the option of storing `kwargs` (even for required arguments). An additional advantage of this is that it makes the ASDF storage just a little easier.